modules/core/message_list_functions.php

```diff
@@ -206,7 +206,18 @@ function message_list_row($values, $id, $style, $output_mod, $row_class='') {
     if ($row_class) {
         $res .= ' '.$output_mod->html_safe($row_class);
     }
-    $res .= '">';
+    $data_uid = "";
+    if ($uids = explode("_", $output_mod->html_safe($id))) {
```

Comment:
No need to html_safe this when you are using it in PHP context - like passing it to explode function. Use html_safe when you append output to a template as html content.